broker/setup_broker/swift/setup_broker.sh

```diff
@@ -72,55 +70,95 @@ manage_resources() {
     fi
 
     if [ "$mode" = "setup" ]; then
-        # create BigQuery dataset and table
+        #--- Create BigQuery dataset and table
         echo
         echo "Creating BigQuery dataset and table..."
-        bq --location="${region}" mk --dataset "${bq_dataset_alerts}"
-
-        (cd templates && bq mk --table "${PROJECT_ID}:${bq_dataset_alerts}.${alerts_table}" "bq_${survey}_${alerts_table}_schema.json") || exit 5
-        bq update --description "Alert data from Swift/BAT-GUANO. This table is an archive of the swift-alerts Pub/Sub stream. It has the same schema as the original alert bytes, including nested and repeated fields." "${PROJECT_ID}:${bq_dataset_alerts}.${alerts_table}"
-
-        # create broker bucket and upload files
+        if ! bq ls "${PROJECT_ID}:${bq_dataset}" >/dev/null 2>&1; then
+            bq --location="${region}" mk --dataset "${bq_dataset}"
+            # grant public access to the dataset; for more information, see:
+            # https://cloud.google.com/bigquery/docs/control-access-to-resources-iam#grant_access_to_a_dataset
+            (cd templates && bq update --source "bq_${survey}_policy.json" "${PROJECT_ID}:${bq_dataset}") || exit 5
```

Comment:
Move this to the same block that grants public access to the pubsub topics so that it's only done for production resources.